packages/vm/src/errors.rs

```diff
@@ -204,6 +211,66 @@ pub fn make_uninitialized_context_data<S: Into<String>>(kind: S) -> VmError {
     UninitializedContextData { kind: kind.into() }.build()
 }
 
+#[derive(Debug, Snafu)]
+pub enum FfiError {
+    #[snafu(display("Panic in FFI call"))]
+    ForeignPanic { backtrace: snafu::Backtrace },
+    #[snafu(display("bad argument passed to FFI"))]
+    BadArgument { backtrace: snafu::Backtrace },
+    #[snafu(display("Ran out of gas during FFI call"))]
+    OutOfGas,
+    #[snafu(display("Error during FFI call: {}", error))]
+    Other {
+        error: String,
+        backtrace: snafu::Backtrace,
+    },
+}
+
+impl FfiError {
```

Comment:
What is this for? I generally don't like the idea of modifying errors.
Maybe just a simpler constructor `ffi_error(message: S) -> FfiError` that returns an `FfiError::Other` variant?

Reply:
Oh, this isn't a constructor. you can see how i use it, it's kind of like the `context` method from `snafu::ResultExt`. it basically says "If the value is a specific variant of the type, please give it this additional data". In this case i say that if it can, please add this message with context information. It's used to give a specific error message when go has an unexpected error (not panic) and we have something useful to say.

Reply:
Can this be implemented as a constructor, i.e. keep the error immutable? Or do we get the error from somewhere and need to add the information later on?

Reply:
Uhh, i suppose it could be? Let me play around with it. If you look at the places this is used, we first convert a `GoResult` to `Result<(), FfiError>`, and then if we got `Other` we add an error message with this method. I can wrap that up into an associated function on `GoResult`, perhaps.

Reply:
Not too important in my opinion. Wrote it down here: https://github.com/CosmWasm/cosmwasm/issues/336